src/selectors/operators/numeric/BinaryNumericOperator.js

```diff
@@ -58,32 +61,77 @@ class BinaryNumericOperator extends Selector {
+			}
 		}
-
-		if (!isInstanceOfType(firstValue, 'xs:numeric') || !isInstanceOfType(secondValue, 'xs:numeric')) {
-			// TODO: date / time like values
-			throw new Error('XPTY0004: the operands of the "' + this._kind + '" operator should be of type xs:numeric?.');
+		if (((isInstanceOfType(firstValue, 'xs:yearMonthDuration') && isInstanceOfType(secondValue, 'xs:numeric')) ||
+			(isInstanceOfType(firstValue, 'xs:numeric') && isInstanceOfType(secondValue, 'xs:yearMonthDuration'))) &&
+			this._kind === '*') {
+			typedResult = createAtomicValue(YearMonthDuration.multiply(firstValue.value, secondValue.value), 'xs:yearMonthDuration');
+		}
+		if (isInstanceOfType(firstValue, 'xs:yearMonthDuration') && isInstanceOfType(secondValue, 'xs:numeric') &&
+			this._kind === 'div') {
+			typedResult = createAtomicValue(YearMonthDuration.divide(firstValue.value, secondValue.value), 'xs:yearMonthDuration');
 		}
 
-		const result = executeOperator(this._kind, firstValue.value, secondValue.value);
-		let typedResult;
-		// Override for types
-		if (this._kind === 'div') {
-			typedResult = createAtomicValue(result, 'xs:decimal');
+		if (isInstanceOfType(firstValue, 'xs:dayTimeDuration') && isInstanceOfType(secondValue, 'xs:dayTimeDuration')) {
+			if (this._kind === '+') {
+				typedResult = createAtomicValue(DayTimeDuration.add(firstValue.value, secondValue.value), 'xs:dayTimeDuration');
+			}
+			if (this._kind === '-') {
+				typedResult = createAtomicValue(DayTimeDuration.subtract(firstValue.value, secondValue.value), 'xs:dayTimeDuration');
+			}
+			if (this._kind === 'div') {
+				typedResult = createAtomicValue(DayTimeDuration.divideByDayTimeDuration(firstValue.value, secondValue.value), 'xs:decimal');
+			}
 		}
-		else if (this._kind === 'idiv') {
-			typedResult = createAtomicValue(result, 'xs:integer');
+		if (((isInstanceOfType(firstValue, 'xs:dayTimeDuration') && isInstanceOfType(secondValue, 'xs:numeric')) ||
+			(isInstanceOfType(firstValue, 'xs:numeric') && isInstanceOfType(secondValue, 'xs:dayTimeDuration'))) &&
+			this._kind === '*') {
+			typedResult = createAtomicValue(DayTimeDuration.multiply(firstValue.value, secondValue.value), 'xs:dayTimeDuration');
 		}
-		else {
-			// For now, always return a decimal, it's all the same in JavaScript
-			typedResult = createAtomicValue(result, 'xs:decimal');
+		if (isInstanceOfType(firstValue, 'xs:dayTimeDuration') && isInstanceOfType(secondValue, 'xs:numeric') &&
+			this._kind === 'div') {
+			typedResult = createAtomicValue(DayTimeDuration.divide(firstValue.value, secondValue.value), 'xs:dayTimeDuration');
 		}
+
+		if (!typedResult) {
+			if (isInstanceOfType(firstValue, 'xs:untypedAtomic')) {
+				firstValue = castToType(firstValue, 'xs:double');
+			}
+
+			if (isInstanceOfType(secondValue, 'xs:untypedAtomic')) {
+				secondValue = castToType(secondValue, 'xs:double');
+			}
+
+			if (!isInstanceOfType(firstValue, 'xs:numeric') || !isInstanceOfType(secondValue, 'xs:numeric')) {
```

Comment:
Numerics wil occur way more, could you pull this to the top?
This file will have to be rewritten in the rebase (assuming #6 will land earlier)